packages/blocks/src/api/factory.js

```diff
@@ -134,13 +100,15 @@ export function createBlocksFromInnerBlocksTemplate(
 export function cloneBlock( block, mergeAttributes = {}, newInnerBlocks ) {
 	const clientId = uuid();
 
+	const sanitizedAttributes = sanitizeBlockAttributes( block.name, {
+		...block.attributes,
+		...mergeAttributes,
+	} );
+
 	return {
 		...block,
 		clientId,
-		attributes: {
-			...block.attributes,
-			...mergeAttributes,
-		},
+		attributes: sanitizedAttributes,
```

Comment:
Could you please add a new unit test for `cloneBlock()` which tests that attributes are sanitised? (And, maybe while you're at it, double check that `createBlock()` has one.)

Reply:
Good idea!